src/rgw/rgw_compression.cc

```diff
@@ -0,0 +1,164 @@
+
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "rgw_compression.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+//------------RGWPutObj_Compress---------------
+
+int RGWPutObj_Compress::handle_data(bufferlist& bl, off_t ofs, void **phandle, rgw_obj *pobj, bool *again)
+{
+  bufferlist in_bl;
+
+  // compression stuff
+  if ((ofs > 0 && compressed) ||                                // if previous part was compressed
+      (ofs == 0)) {                                             // or it's the first part
+    ldout(cct, 10) << "Compression for rgw is enabled, compress part " << bl.length() << dendl;
+    CompressorRef compressor = Compressor::create(cct, cct->_conf->rgw_compression_type);
+    if (!compressor.get()) {
+      if (ofs > 0 && compressed) {
+        lderr(cct) << "Cannot load compressor of type " << cct->_conf->rgw_compression_type
+                            << " for next part, compression process failed" << dendl;
+        return -EIO;
+      }
+      // if compressor isn't available - just do not use it with log warning?
+      ldout(cct, 5) << "Cannot load compressor of type " << cct->_conf->rgw_compression_type 
+                    << "for rgw, check rgw_compression_type config option" << dendl;
```

Comment:
missing space in output: `<< " for rgw`. this was copy-pasted twice below in `RGWGetObj_Decompress` as well